frontend/selection.tsx

```diff
@@ -60,8 +87,27 @@ export function Selection({
       d.x - d.deltaX - shapeCenter.x
     );
     const after = Math.atan2(offsetY - shapeCenter.y, d.x - shapeCenter.x);
+    rep.mutate.rotateShape({
```

Comment:
Can we factor the repeated math here into some kind of helper? These functions are all exactly the same except for the part that gets multiplied by 180.